orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/MonikerHelper.java

```diff
@@ -58,4 +59,28 @@ static public Moniker monikerFromStage(Stage stage) {
       return null;
     }
   }
+
+  static public Moniker monikerFromStage(Stage stage, String fallbackFriggaName) {
+    Moniker moniker = monikerFromStage(stage);
+    return monikerOrFrigga(moniker, fallbackFriggaName);
+  }
+
+  static public Moniker monikerOrFrigga(Moniker moniker, String friggaName) {
+    if (moniker == null) {
```

Comment:
If interested, a one liner equivalent: `Optional.ofNullable(moniker).orElse(friggerToMoniker(friggaName)`

Reply:
Or `moniker == null ? friggaToMoniker(friggaName) : moniker`

Reply:
That’s even better.